### ISSUE TITLE
Use `np.array` instead of `np.ndarray`

### ISSUE THREAD

**Issue by eywalker** [datajoint/datajoint-python#184]
I've noticed that there are multiple places where `np.ndarray` object constructor is used instead of `np.array`. As the [official documentation states](http://docs.scipy.org/doc/numpy-1.10.0/reference/generated/numpy.ndarray.html), direct use of `np.ndarray` is discouraged. Unless there is a good reason that we should be using constructor directly, we should use `np.array` or other methods instead. Quick inspection yields that the most common use of this seems to be in selecting out sub fields of the structure/record array. This should be done via

```python
# ['name', 'age', 'gender', 'DOB', 'nationality'] be the name of fields of record array `data`
# following syntax selects out specified fields only
sub_data = data[['name', 'age']].copy() 
```

Use of `np.ndarray` requires specification of unnecessary values like `offset` and `strides` (which is ultimately not used) and makes it difficult to understand the code.

**Comment by fabiansinz** [comment]
I agree.

**Comment by eywalker** [comment]
`np.ndarray` is used just to select out sub fields in the following places:

* `fetch.py`: `fetch.__getitem__` and `fetch1.__getitem__`
* `utils.py`: `groupby`

Since I didn't write any of these directly, I'd feel more comfortable asking the original authors (I'm guessing @fabiansinz  for the `fetch` methods and @dimitri-yatsenko for `groupby`) to make the respective changes :)

**Comment by fabiansinz** [comment]
Nope, both of them originate from me. Isn't it nice if you agree to changes for which you yourself are the culprit. I'll fix it.

**Comment by dimitri-yatsenko** [comment]
*like*

**Comment by eywalker** [comment]
Haha, thanks fabee!

**Comment by dimitri-yatsenko** [comment]
Thank you for pointing this out.  Fixed in 9841506.
